association-backend/src/webhooks/auth_backend.py

```diff
@@ -0,0 +1,38 @@
+import base64
+import binascii
+
+from starlette.authentication import (
+    AuthCredentials,
+    AuthenticationBackend,
+    AuthenticationError,
+    SimpleUser,
+)
+
+from src.association.settings import PRETIX_WEBHOOK_SECRET
+
+
+class PretixAuthBackend(AuthenticationBackend):
+    async def authenticate(self, request):
+        if request.url.path != "/pretix-webhook":
+            return
```

Comment:
should we raise an error just in case? :)

Reply:
In an ideal world returning None should mean "go to the next authentication backend" so returning an error feels wrong 🤔  i.e see https://www.starlette.io/authentication/ when the header is missing

In the real world, this doesn't matter because this auth backend never gets invoked when it isn't pretix-webhook since we do the routing here: https://github.com/pythonitalia/pycon/blob/aa33ea5f78a396b7d8fe370c4c6ca6f4d2c518b7/association-backend/src/association/auth.py#L15


I would keep the `return` just because it is similar to what behaviour Starlette recommends in their doc, and if in the future they add support for multiple backends it should work out of the box (this is just guessing, since who knows what will happen in X months/years)

Reply:
Changed this at the end because I want to make sure `PretixAuthBackend` is not used in some other way